x/ibc/testing/coordinator.go

```diff
@@ -0,0 +1,178 @@
+package testing
+
+import (
+	"testing"
+	"time"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	connectiontypes "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+)
+
+var (
+	ClientIDPrefix  = "clientIDForChain"
+	globalStartTime = time.Date(2020, 1, 2, 0, 0, 0, 0, time.UTC)
+	timeIncrement   = time.Second * 5
+)
+
+// Coordinator is a testing struct which contains N TestChain's. It handles keeping all chains
+// in sync with regards to time.
+type Coordinator struct {
+	Chains []*TestChain
```

Comment:
I'd also add a priv field to track the index in the array from a given ChainID: `map[string]int` (i.e `map[chainID]idx`). Then you can use chainIDs for each of the functions which seems more intuitive to me.

Reply:
I agree that indexes are not the best to work with. Maybe just having a `map[string]*TestChain`?

Reply:
I think turning the `ClientID` field to `ChainID` (and making a client id array) makes sense with respect to this mapping.